ehcache-impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -104,6 +110,18 @@ private void internalStart() {
       if (lock == null) {
         throw new RuntimeException("Persistence directory already locked by another process: " + rootDirectory.getAbsolutePath());
       }
+
+      if (cleanFile.exists()) {
+        try {
+          LOGGER.debug("clean file exists, trying to delete the file.");
+          Files.delete(cleanFile.toPath());
+          clean = true;
+          LOGGER.debug("clean file is deleted.");
+        } catch (IOException e) {
+          LOGGER.warn("clean file was not deleted {}.", cleanFile.getPath());
```

Comment:
I'm thinking this should be a hard failure. If the clean file isn't deleted here then a subsequent dirty start would be seen as clean.